File: tools/wave/src/utils/user-agent-parser.js
const useragent = require("useragent");

class UserAgentParser {
  parse(ua) {
    const parsed = useragent.parse(ua);
    if (parsed.family === "Apple Mail") {
      parsed.family = "WebKit";
    }
    return {
      browser: {
        name: parsed.family || "unknown",
        version: parsed.major || "00"
      },
      os: {
        name: parsed.os.family
      },
      device: {
        name: parsed.device.family
      }
    };
  }

  abbreviateBrowserName(name) {
    const shortnames = {
      Chrome: "Ch",
      "Chrome Mobile WebView": "Ch",
      Chromium: "Cm",
      WebKit: "Wk", // WebKit build
      Safari: "Sf",
      Firefox: "FF",
      IE: "IE",
      Edge: "Ed",
      Opera: "Op"
    };
    let short = shortnames[name];
    return short || "Xx";
  }
}

const userAgentParser = new UserAgentParser();

module.exports = userAgentParser;

File: tools/wptserve/wptserve/wave/utils/user_agent_parser.py
from __future__ import absolute_import
from ua_parser import user_agent_parser


def parse_user_agent(user_agent_string):
    user_agent = user_agent_parser.ParseUserAgent(user_agent_string)

    name = user_agent[u"family"]
    version = u"0"

    if user_agent[u"major"] is not None:
        version = user_agent[u"major"]

    if user_agent[u"minor"] is not None:
        version = version + u"." + user_agent[u"minor"]

    if user_agent[u"patch"] is not None:
        version = version + u"." + user_agent[u"patch"]

    return {
        u"name": name,
        u"version": version
    }
